Source/WebCore/css/process-css-pseudo-selectors.py

```diff
@@ -626,75 +614,68 @@ def write_is_pseudo_element_enabled(self, writer, values):
 
-        shadow_map = {}
-        shadow_alias_map = {}
+        user_agent_part_map = {}
+        user_agent_part_alias_map = {}
 
-        # Collect shadow pseudo elements and generate code for non-shadow pseudo elements.
+        # Collect user agent parts and generate code for other pseudo-elements.
         for pseudo_name, pseudo_data in values.items():
             is_internal_pseudo = pseudo_name.startswith('-internal-')
             if 'settings-flag' not in pseudo_data and not is_internal_pseudo:
                 continue
 
-            settings_flag = pseudo_data['settings-flag'] if 'settings-flag' in pseudo_data else None
+            settings_flag = pseudo_data.get('settings-flag', None)
             enablement_condition = self.format_enablement_condition(settings_flag, is_internal_pseudo)
 
             if key_is_true(pseudo_data, 'user-agent-part'):
-                if 'condition' in pseudo_data:
-                    shadow_map[pseudo_name] = (enablement_condition, pseudo_data['condition'])
-                else:
-                    shadow_map[pseudo_name] = (enablement_condition, None)
-
-                if 'aliases' in pseudo_data:
-                    for alias in pseudo_data['aliases']:
-                        if 'condition' in pseudo_data:
-                            shadow_alias_map[alias] = (enablement_condition, pseudo_data['condition'])
-                        else:
-                            shadow_alias_map[alias] = (enablement_condition, None)
+                user_agent_part_map[pseudo_name] = (enablement_condition, pseudo_data.get('conditional', None))
+
+                for alias in pseudo_data.get('aliases', []):
+                    user_agent_part_alias_map[alias] = (enablement_condition, pseudo_data.get('conditional', None))
                 continue
 
-            # Generate code for non-shadow pseudo elements.
-            if 'condition' in pseudo_data:
-                writer.write('#if {}'.format(pseudo_data['condition']))
+            # Generate code for pseudo-elements that are not user agent parts.
+            if 'conditional' in pseudo_data:
+                writer.write(f'#if {pseudo_data["conditional"]}')
 
             with writer.indent():
-                writer.write('case PseudoElement::{}:'.format(format_name_for_enum_class(pseudo_name)))
+                writer.write(f'case PseudoElement::{format_name_for_enum_class(pseudo_name)}:')
 
                 with writer.indent():
-                    writer.write('return {};'.format(enablement_condition))
+                    writer.write(f'return {enablement_condition};')
 
-            if 'condition' in pseudo_data:
+            if 'conditional' in pseudo_data:
                 writer.write('#endif')
 
-        # Generate code for shadow pseudo elements.
-        def write_condition_cases_for_shadow_elements(map):
+        # Generate code for user agent parts.
+        def write_condition_cases_for_user_agent_parts(map):
```

Comment:
Maybe write_*conditional*_cases_for_user_agent_parts?

Reply:
Changed it to `write_cases_for_user_agent_parts`